checker.cpp

```diff
@@ -1,17 +1,82 @@
 #include <assert.h>
+#include<iostream>
+using namespace std;
+const float bpmLimit[]={70,150};
+const float spo2Limit[]={90,100};//make INT_MAX
+const float respRateLimit[]={30,95};
+class Alert
+{
+  public:
+    virtual void sendAlert(const char* vitalName,const char* value)=0;//pure virtual fun
+    virtual float showValue(const float val)=0;
+};
 
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
-    return false;
-  } else if(spo2 < 90) {
-    return false;
-  } else if(respRate < 30 || respRate > 95) {
-    return false;
-  }
-  return true;
-}
+class AlertSms:public Alert
+{
+  public:
+    void sendAlert(const char* vitalName,const char* value) override{
+      cout<<"SMS:"<<vitalName<<"--------"<<value<<endl;}
+    float showValue(const float val) override{
+      cout<<"Value;---:"<<val<<endl;
+    }
+};
+
+class AlertSound:public Alert
+{
+  public:
+    void sendAlert(const char* vitalName,const char* value) override{
+      cout<<"SOUND:"<<vitalName<<"--------"<<value<<endl;}
+     float showValue(const float val) override{
+      cout<<"Value;---:"<<val<<endl;
+    } 
+};
+
+class vitals
+{
+ public:
+    const char* vitalIsOk(float value,const float *limits)
+    {
+      if(value< limits[0]){return "below";}
+      else if(value> limits[1]){return "above";}
+      else {return "all okay";}
+    }
+    
+    void vitalsAreOk(float bpm, float spo2, float respRate) 
+    {
+       // int bool_array[]={int(vitalIsOk(bpm,bpmLimit)),int(vitalIsOk(spo2,spo2Limit)),int(vitalIsOk(respRate,respRateLimit))};
+        //if(int(vitalIsOk(bpm,bpmLimit)) +int( vitalIsOk(spo2,spo2Limit)) +int(vitalIsOk(respRate,respRateLimit))!=3){return false;}
+        //else return true;
+        const char* bpmlevel=vitalIsOk(bpm,bpmLimit);
+        const char* spo2level=vitalIsOk(spo2,spo2Limit);
+        const char* respRatelevel=vitalIsOk(respRate,respRateLimit);
+        //alerting
+        Alert *alert;
+        AlertSms alertSms;
+        AlertSound alertSound;
+        //derv class addr pass to bas class pointer
+        alert=&alertSms;
+        alert->sendAlert("BPM",bpmlevel);
+        alert->showValue(bpm);
```

Comment:
Normally an 'alert' is not required if the vitals are ok, to prevent staff from being overloaded with messages. How would you avoid an 'all ok' alert in your structure?